flight/Modules/System/systemmod.c

```diff
@@ -67,14 +65,45 @@
 #if defined(PIOS_SYSTEM_STACK_SIZE)
 #define STACK_SIZE_BYTES PIOS_SYSTEM_STACK_SIZE
 #else
-#define STACK_SIZE_BYTES 924
+#define STACK_SIZE_BYTES 1024
 #endif
 
-#define TASK_PRIORITY PIOS_THREAD_PRIO_LOW
+#define TASK_PRIORITY PIOS_THREAD_PRIO_NORMAL
+
+/* Generates an armed LED blink of 4.4Hz, close to the previous 5, and is
+ * a nice prime number to not generate as nasty beat frequencies as other
+ * choices */
+#define SYSTEM_UPDATE_PERIOD_MS 113
+
+// Private types
+
+/**
+ * Event callback information
+ */
+typedef struct {
+	UAVObjEvent ev; /** The actual event */
+	UAVObjEventCallback cb; /** The callback function, or zero if none */
+	struct pios_queue *queue; /** The queue or zero if none */
+} EventCallbackInfo;
+
+/**
+ * List of object properties that are needed for the periodic updates.
+ */
+struct PeriodicObjectListStruct {
+	EventCallbackInfo evInfo; /** Event callback information */
```

Comment:
All of this is moved as originally indented for clarity.  Can be reindented later.